base/strings/types.jl

```diff
@@ -10,26 +10,34 @@ struct SubString{T<:AbstractString} <: AbstractString
     endof::Int
 
     function SubString{T}(s::T, i::Int, j::Int) where T<:AbstractString
-        if i > endof(s) || j<i
-            return new(s, i-1, 0)
-        else
-            if !isvalid(s,i)
-                throw(ArgumentError("invalid SubString index"))
-            end
-
-            while !isvalid(s,j) && j > i
-                j -= 1
-            end
-
-            o = i-1
-            new(s, o, max(0, j-o))
+        i > j && return new(s, 0, 0)
+        i < 1 && throw(BoundsError(s, i))
+        j > endof(s) && throw(BoundsError(s, j))
+
+        if !isvalid(s,i)
+            throw(ArgumentError("invalid SubString index i"))
```

Comment:
I meant symbol `i` as it is a name of the first argument. Invalid second argument `j` is allowed (like in `getinex`).